plugins/readtimer.koplugin/main.lua

```diff
@@ -66,26 +125,109 @@ function ReadTimer:remaining()
     end
 end
 
-function ReadTimer:remainingTime()
+-- can round
+function ReadTimer:remainingTime(round)
     if self:scheduled() then
         local remainder = self:remaining()
+        if round then
+            if round < 0 then -- round down
+                remainder = remainder - 59
+            elseif round == 0 then
+                remainder = remainder + 30
+            else -- round up
+                remainder = remainder + 59
+            end
+            remainder = math.floor(remainder * (1/60)) * 60
+        end
+
         local hours = math.floor(remainder * (1/3600))
         local minutes = math.floor(remainder % 3600 * (1/60))
         local seconds = math.floor(remainder % 60)
         return hours, minutes, seconds
     end
 end
 
+function ReadTimer:addAdditionalHeaderContent()
+    self.ui.crelistener:addAdditionalHeaderContent(self.additional_header_content_func)
+    self.update_status_bars(self, -1)
+end
+function ReadTimer:addAdditionalFooterContent()
+    self.ui.view.footer:addAdditionalFooterContent(self.additional_footer_content_func)
+    self.update_status_bars(self, -1)
+end
+
+function ReadTimer:removeAdditionalHeaderContent()
+    self.ui.crelistener:removeAdditionalHeaderContent(self.additional_header_content_func)
+    self.update_status_bars(self, -1)
+    UIManager:broadcastEvent(Event:new("UpdateHeader"))
+end
+function ReadTimer:removeAdditionalFooterContent()
+    self.ui.view.footer:removeAdditionalFooterContent(self.additional_footer_content_func)
+    self.update_status_bars(self, -1)
+    UIManager:broadcastEvent(Event:new("UpdateFooter", true))
+end
+
 function ReadTimer:unschedule()
     if self:scheduled() then
         UIManager:unschedule(self.alarm_callback)
+        UIManager:unschedule(self.update_status_bars, self)
```

Comment:
Haven't looked how this is cleaned when leaving Reader - but be sure the unscheduling is always done, we don't want it to be called 55 seconds later with `self` gone.